packages/gatsby/src/utils/webpack.config.js

```diff
@@ -491,6 +504,7 @@ module.exports = async (
           `@gatsbyjs/webpack-hot-middleware`
         ),
         $virtual: getAbsolutePathForVirtualModule(`$virtual`),
+        webpack: getPackageRoot(`webpack`),
```

Comment:
Dedupe `webpack` package - `gatsby` has webpack dependency and now `@rspack/dev-client` also has `webpack` dependency. If package manager chose to install 2 instances, that breaks HMR. So we use alias here to "dedupe" it in compiled bundle